mir_planning/mir_actions/mir_manipulate_drawer/ros/scripts/close_drawer_server

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+import rospy
+import smach
+
+# import of generic states
+import mir_states.common.manipulation_states as gms
+import mcr_states.common.basic_states as gbs
+import mir_states.common.action_states as gas
+
+# action lib
+from smach_ros import ActionServerWrapper, IntrospectionServer
+
+from mir_planning_msgs.msg import GenericExecuteAction, GenericExecuteResult, GenericExecuteFeedback
+from mir_actions.utils import Utils
+from mir_manipulate_drawer.manipulate_drawer_utils import ManipulateDrawerUtils
+from mir_manipulate_drawer.manipulate_drawer_utils import Setup, SendSafePoseToDBC, SendPoseToDBC
+        
+def main():
+    # Open the container
+    rospy.init_node('close_drawer_server')
+    # Construct state machine
+    sm = smach.StateMachine(
+            outcomes=['OVERALL_SUCCESS','OVERALL_FAILED'],
+            input_keys = ['goal'],
+            output_keys = ['feedback', 'result'])
+
+    with sm:
+        # ASSUMPTION: the robot is aligned perpendicular to an open drawer
+        smach.StateMachine.add(
+                'SETUP',
+                Setup(),
+                transitions={'succeeded':'PUBLISH_TEMP_REFERENCE_FRAME_BACKWARD_MOTION'})
+        
+        # publish a temporary reference frame for moving backward if necessary
+        smach.StateMachine.add(
+                'PUBLISH_TEMP_REFERENCE_FRAME_BACKWARD_MOTION',
+                gbs.send_event([('/static_transform_publisher_node/event_in', 'e_start')]),
+                transitions={'success':'SET_DBC_PARAMS'})
+
+        smach.StateMachine.add(
+                'SET_DBC_PARAMS',
+                gbs.set_named_config('dbc_pick_object'),
+                transitions={'success': 'SEND_SAFE_POSE_TO_DBC',
+                             'timeout': 'OVERALL_FAILED',
+                             'failure': 'OVERALL_FAILED'})
+
+        # - Move base back for perceive drawer handle
+        smach.StateMachine.add(
+                'SEND_SAFE_POSE_TO_DBC',
+                SendSafePoseToDBC(target_dist=0.42),
```

Comment:
think about how to calculate this parameter (using expected position of end effector in the target arm pose)

Reply:
https://github.com/b-it-bots/mas_industrial_robotics/issues/103